### DISCUSSION TITLE
Deploy contracts using mobile or browser wallets

### DISCUSSION THREAD

**Discussion by escottalexander** [scaffold-eth/scaffold-eth-2#868] [Ideas]
The reliance on using a local file containing a seed phrase (whether generated or pasted from your wallet) in order to deploy contracts is untenable as it ends up with users accidentally committing their seed phrase and generally encouraging bad opsec. Optimism's recent Onchain Builders Retro Funding initiative pushed for builders to prove themselves as the holders of the deployer for their contracts. This highlights the disconnect between using a throwaway deployer account with a lower security threshold and using an account that you intend to keep safe (and remember) long term.

Additionally, Patrick Collins recently [posted](https://x.com/PatrickAlphaC/status/1798465648856891682) about the frameworks culpability in newbie devs getting hacked when we push bad practices like this.

**Comment by escottalexander** [comment]
Another option that would be better in some ways/worse in others is if you were prompted for your seed phrase (or pk) and it was just used for the deployment process - not saved anywhere. Spin on this would be saving the seed phrase in an encrypted state and prompting for the password when you deploy.

**Comment by carletex** [comment]
Related: https://github.com/scaffold-eth/scaffold-eth-2/issues/812 I think we should implement it when doing https://github.com/scaffold-eth/scaffold-eth-2/issues/866

For hardhat, we could even implement a simple system to encrypt/decrypt your PK with a password?

Anyway, 100% that we should improve PK safety.

**Reply by escottalexander** [reply]
Awesome! Did not realize that issue existed.
